pkg/registry/core/pod/storage/storage_test.go

```diff
@@ -744,6 +745,76 @@ func TestEtcdCreateWithConflict(t *testing.T) {
 	}
 }
 
+func validNewBinding() *api.Binding {
+	return &api.Binding{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+		Target:     api.ObjectReference{Name: "machine", Kind: "Node"},
+	}
+}
+
+func TestEtcdCreateBindingWithUID(t *testing.T) {
+	testCases := map[string]struct {
+		podUIDGetter     func(pod *api.Pod) types.UID
+		errOK            func(error) bool
+		expectedNodeName string
+	}{
+		"originUID": {
+			podUIDGetter: func(pod *api.Pod) types.UID {
+				return pod.UID
+			},
+			errOK: func(err error) bool {
+				return err == nil
+			},
+			expectedNodeName: "machine",
+		},
+		"emptyUID": {
+			podUIDGetter: func(pod *api.Pod) types.UID {
+				return ""
+			},
+			errOK: func(err error) bool {
+				return err == nil
+			},
+			expectedNodeName: "machine",
+		},
+		"changedUID": {
+			podUIDGetter: func(pod *api.Pod) types.UID {
+				return pod.UID + "-changed"
+			},
+			errOK: func(err error) bool {
+				return err != nil && errors.IsConflict(err)
+			},
+			expectedNodeName: "",
+		},
+	}
+
+	storage, bindingStorage, _, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.Store.DestroyFunc()
+
+	for k, testCase := range testCases {
+		pod := validNewPod()
+		pod.Namespace = fmt.Sprintf("namespace-%s", strings.ToLower(k))
+		ctx := genericapirequest.WithNamespace(genericapirequest.NewDefaultContext(), pod.Namespace)
+
+		if _, err := storage.Create(ctx, pod, rest.ValidateAllObjectFunc, &metav1.CreateOptions{}); err != nil {
+			t.Fatalf("%s: unexpected error: %v", k, err)
+		}
+
+		binding := validNewBinding()
+		binding.UID = testCase.podUIDGetter(pod)
```

Comment:
the pod passed in here is not the result of the create... I suspect it does not have a UID set, so the originUID case is not exercising what you think it is

Reply:
> the pod passed in here is not the result of the create... I suspect it does not have a UID set, so the originUID case is not exercising what you think it is

I use it by mistake since `storage.Create` here will mutate the input (set UID in `pod`). but the originUID case is exercising what i think it is.....however, it's a mistake. I've fixed it in the new commit.

@liggitt could you please help to review again :-)